dev/bots/analyze.dart

```diff
@@ -124,11 +125,100 @@ Future<void> run(List<String> arguments) async {
   }
 }
 
+
+// TESTS
+
+final RegExp _findDeprecationPattern = RegExp(r'@[Dd]eprecated');
+final RegExp _deprecationPattern1 = RegExp(r'^( *)@Deprecated\($'); // ignore: flutter_deprecation_syntax (see analyze.dart)
+final RegExp _deprecationPattern2 = RegExp(r"^ *'(.+) '$");
+final RegExp _deprecationPattern3 = RegExp(r"^ *'This feature was deprecated after v([0-9]+)\.([0-9]+)\.([0-9]+)\.(?: See: (https://flutter.dev/.+))?'$");
+final RegExp _deprecationPattern4 = RegExp(r'^ *\)$');
+
+/// Some deprecation notices are special, for example they're used to annotate members that
+/// will never go away and were never allowed but which we are trying to show messages for.
+/// (One example would be a library that intentionally conflicts with a member in another
+/// library to indicate that it is incompatible with that other library. Another would be
+/// the regexp just above...)
+const String _ignoreDeprecation = ' // ignore: flutter_deprecation_syntax (see analyze.dart)';
+
+/// Some deprecation notices are grand-fathered in for now. They must have an issue listed.
+final RegExp _grandfatheredDeprecation = RegExp(r' // ignore: flutter_deprecation_syntax, https://github.com/flutter/flutter/issues/[0-9]+$');
+
+Future<void> verifyDeprecations(String workingDirectory) async {
+  final List<String> errors = <String>[];
+  for (File file in _dartFiles(workingDirectory)) {
+    int lineNumber = 0;
+    final List<String> lines = file.readAsLinesSync();
+    final List<int> linesWithDeprecations = <int>[];
+    for (String line in lines) {
+      if (line.contains(_findDeprecationPattern) &&
+          !line.endsWith(_ignoreDeprecation) &&
+          !line.contains(_grandfatheredDeprecation)) {
+        linesWithDeprecations.add(lineNumber);
+      }
+      lineNumber += 1;
```

Comment:
```suggestion
      lineNumber++;
```

Reply:
We prefer `+=` over `++` for several reasons. First, in some languages/compilers suffix `++` is an antipattern because of performance reasons, and so it's easier to just avoid it in general. Second, it isn't as obvious that it actually mutates the underlying variable, whereas `+=` very clearly does (it's an assignment with an `=` sign). Third, it makes it more convenient when changing the increment to a number other than 1.